dc/s2n-quic-dc/src/stream/client/rpc.rs

```diff
@@ -44,10 +49,20 @@ where
             }
         }
     };
+    let mut reader = core::pin::pin!(reader);
 
-    let (_, out) = ::tokio::try_join!(writer, reader)?;
+    poll_fn(|cx| {
+        // Poll the `writer` as long as it hasn't completed or the `reader` is still being polled
+        if !writer_finished {
+            writer_finished = writer.as_mut().poll(cx)?.is_ready();
+        }
```

Comment:
Are we worried about not clearing a socket readiness state that causes e.g. level-triggered notifications to keep firing? I think that's not an issue with mio on Linux, but not sure if we'd run into trouble on (say) Windows.

Reply:
Not sure I follow the question. If the reader finishes before the writer does, then it'll trigger the reader's drop implementation, which should notify the background task that the application has detached and it will drive it to completion.